### ISSUE TITLE
Set global attributes for edge or node

### ISSUE THREAD

**Issue by cakraww** [emicklei/dot#15]
How to set node/edge attribute globally like in:
```
digraph G
{
    node[shape="point"];
    edge[arrowhead="odiamond"];

    A -> B
    ...
}
```
so that we don't have to set the attribute on each individual node/edge?

**Comment by emicklei** [comment]
atm, there is no convenient way to achieve this.  You could create wrapper methods to keep DRY.

**Comment by emicklei** [comment]
this could be achieved with a graph option to be created.
NewGraph(DefaultNodeAttribute("shape","point"), DefaultEdgeAttribute("arrowhead","odiamond"))

**Comment by emicklei** [comment]
i am closing this now. feel free to reopen if someone else shows interest

**Comment by neomagus00** [comment]
+1 to this! My use case is setting "decorate=true" on all node labels.

**Comment by emicklei** [comment]
please have a look at this proposed extension

**Comment by kevin-peirce** [comment]
+1 for me too. I want to do stuff like this in the output DOT file:

```
        /* Common attributes for nodes and edges */
        node [
            shape=rectangle,
            fontname="arial",
            style="rounded,filled"
        ]

        edge [
            fontname=arial,
            fontsize=9, 
            arrowsize=0.8,
            arrowhead=open
        ]
```

**Comment by kevin-peirce** [comment]
> +1 for me too. I want to do stuff like this in the output DOT file:
> 
> ```
>         /* Common attributes for nodes and edges */
>         node [
>             shape=rectangle,
>             fontname="arial",
>             style="rounded,filled"
>         ]
> 
>         edge [
>             fontname=arial,
>             fontsize=9, 
>             arrowsize=0.8,
>             arrowhead=open
>         ]
> ```

Just saw you'd done a proposed change - works a treat for me thanks!  For me it looked like this:

```
	di.NodeInitializer(func(n dot.Node) {
		n.Attr("shape", "rectangle")
		n.Attr("fontname", "arial")
		n.Attr("style", "rounded,filled")
	})

	di.EdgeInitializer(func(e dot.Edge) {
		e.Attr("fontname", "arial")
		e.Attr("fontsize", "9")
		e.Attr("arrowsize", "0.8")
		e.Attr("arrowhead", "open")
	})
```

**Comment by emicklei** [comment]
this change has been merged in v0.15.0
